python/ray/serve/deployment_state.py

```diff
@@ -246,6 +248,39 @@ def start(self, deployment_info: DeploymentInfo, version: DeploymentVersion):
             self._controller_namespace,
             self._detached,
         )
+        # TODO(simon): unify the constructor arguments across language
```

Comment:
what is blocking doing this today? would prefer to avoid these independent codepaths